pkg/prove/proof_test.go

```diff
@@ -70,6 +75,151 @@ func TestTxInclusion(t *testing.T) {
 	}
 }
 
+func TestShareInclusion(t *testing.T) {
+	blobs := append(
+		testfactory.GenerateBlobsWithNamespace(
+			100,
+			500,
+			[]byte{0, 0, 0, 0, 0, 1, 0, 0},
+		),
+		append(
+			testfactory.GenerateBlobsWithNamespace(
+				50,
+				500,
+				[]byte{0, 0, 0, 1, 0, 0, 0, 0},
+			),
+			testfactory.GenerateBlobsWithNamespace(
+				50,
+				500,
+				[]byte{0, 0, 1, 0, 0, 0, 0, 0},
+			)...,
+		)...,
+	)
+	sort.Sort(blobs)
+	blockData := types.Data{
+		Txs:        testfactory.GenerateRandomTxs(50, 500),
+		Blobs:      blobs,
+		SquareSize: 32,
+	}
+
+	// not setting useShareIndexes because the transactions indexes do not refer
+	// to the messages because the square and transactions were created manually.
+	rawShares, err := shares.Split(blockData, false)
+	if err != nil {
+		panic(err)
+	}
+
+	// erasure the data square which we use to create the data root.
+	eds, err := da.ExtendShares(blockData.SquareSize, shares.ToBytes(rawShares))
+	require.NoError(t, err)
+
+	// create the new data root by creating the data availability header (merkle
+	// roots of each row and col of the erasure data).
+	dah := da.NewDataAvailabilityHeader(eds)
+	dataRoot := dah.Hash()
+
+	type test struct {
+		name          string
+		startingShare int64
+		endingShare   int64
+		namespaceID   nmtnamespace.ID
+		shouldPass    bool
```

Comment:
[optional] for consistency with other tests in this repo
```suggestion
		expectErr    bool
```

Reply:
4595dd2